### DISCUSSION TITLE
extractingBigDecimal

### DISCUSSION THREAD

**Discussion by michaldo** [assertj/assertj#3077] [Ideas]
Did you consider something like extractingBigDecimal, extractingInstant, to simplify and easy access type-specific assertions? 
Example
as-is
```
record R(BigDecimal bd) {}

assertThat(new R(new BigDecimal("1.1")))
  .extracting(R::bd)
  .isInstanceOfSatisfying(BigDecimal.class, bd -> assertThat(bd).isCloseTo(new BigDecimal("1.099"), withPercentage(1)));
```
nice-to-have
```
assertThat(new R(new BigDecimal("1.1")))
  .extractingBigDecimal(R::bd)
  .isCloseTo(new BigDecimal("1.099"), withPercentage(1)));

**Comment by scordio** [comment]
It's already possible to write the following, relying on [`as`](https://www.javadoc.io/doc/org.assertj/assertj-core/latest/org/assertj/core/api/Assertions.html#as(org.assertj.core.api.InstanceOfAssertFactory)) (syntax sugar) and [`BIG_DECIMAL`](https://www.javadoc.io/doc/org.assertj/assertj-core/latest/org/assertj/core/api/InstanceOfAssertFactories.html#BIG_DECIMAL):

```java
assertThat(new R(new BigDecimal("1.1")))
  .extracting(R::bd, as(BIG_DECIMAL))
  .isCloseTo(new BigDecimal("1.099"), withPercentage(1));
```

[`asInstanceOf`](https://www.javadoc.io/doc/org.assertj/assertj-core/latest/org/assertj/core/api/Assert.html#asInstanceOf(org.assertj.core.api.InstanceOfAssertFactory)) is another way to narrow down the assertion type.

**Reply by midovlvn** [reply]
God bless you
